src/controllers/ssoController.ts

```diff
@@ -58,3 +67,37 @@ export const getUserForBPD = (
       _ => ResponseSuccessJson(_)
     )
   );
+
+// TODO: We need the fn-app client to get the user profile.
+export const getUserForFIMS = (profileService: ProfileService) => (
+  req: express.Request
+): Promise<
+  | IResponseErrorValidation
+  | IResponseErrorInternal
+  | IResponseErrorTooManyRequests
+  | IResponseErrorNotFound
+  | IResponseSuccessJson<FIMSUser>
+> =>
+  withUserFromRequest(req, async user => {
+    const userProfile = await profileService.getProfile(user);
+    if (userProfile.kind !== "IResponseSuccessJson") {
```

Comment:
If you want you can make it more functional taking some ideas from ZendeskController. It will be easier to port it on fp-ts2 with pipes.

Reply:
Fixed in [9682994](https://github.com/pagopa/io-backend/pull/884/commits/9682994fd6ea22ba33c8fb5e3f6076d673ff38e9)